### PR TITLE
Add a flag to fill-in `PSBT_GLOBAL_XPUB`, expose `--offline_signers` in the CLI

### PR DESCRIPTION
**Pull request by afilini**
Still in draft since I have to write tests for `TxBuilder::add_global_xpubs()`

### CODE REVIEW

**Comment by afilini** [comment]
Also needs the CHANGELOG update

**Comment by notmandatory** [comment]
Not a big deal, but I've converted the cli mod to use `structopt` instead of `clap` in my docs PR #194 so will need to make these new cli changes there also. I'll do it after this goes in, or should we put #194 in first and then update this PR to use `structopt`?

**Comment by afilini** [comment]
Maybe we can merge this one first to master and then either rebase `missing-docs` on the new master or just merge `master` in `missing-docs` and during the rebase/merge we can add another commit to add the new options to structopt

**Comment by afilini** [comment]
Should be ready for review. There's an interesting discussion about the correct behavior in issue #199. The implementation should match what was said in that discussion.

**Comment by afilini** [comment]
Rebase on `master` to fix the CI

File: src/wallet/mod.rs [hunk 1]
```diff
@@ -1158,6 +1160,45 @@ where
...
+                    .into_iter()
+                    .map(|child| match child {
+                        ChildNumber::Normal { index } => index.to_le_bytes(),
+                        ChildNumber::Hardened { index } => (index + 0x80000000).to_le_bytes(),
+                    })
+                    .fold(vec![], |mut concat, child| {
+                        concat.extend(&child);
+                        concat
+                    });
+                value.extend(path);
```
Comment (RCasatta) [inline src/wallet/mod.rs]: I think you can use `KeySource::serialize()` instead

File: src/descriptor/mod.rs [hunk 2]
```diff
@@ -339,6 +340,31 @@ impl DescriptorMeta for Descriptor<DescriptorPublicKey> {
...
     }
 
+    fn get_extended_keys(&self) -> Result<Vec<DescriptorXKey<ExtendedPubKey>>, Error> {
+        let get_key = |key: &DescriptorPublicKey,
+                       keys: &mut Vec<DescriptorXKey<ExtendedPubKey>>|
+         -> Result<DummyKey, Error> {
+            match key {
+                DescriptorPublicKey::SinglePub(_) => {}
+                DescriptorPublicKey::XPub(xpub) => keys.push(xpub.clone()),
+            }
```
Comment (RCasatta) [inline src/descriptor/mod.rs]: `if let DescriptorPublicKey::XPub(xpub) = key { ... }` ?

File: src/wallet/mod.rs (hunk 1 above)
Comment (afilini) [reply src/wallet/mod.rs]: That's nice, I didn't know you could serialize it directly. I can't find where they document how the child numbers are serialized,  whether they are encoded as big endian or little endian. I'll try changing the code and if the tests pass I guess it means that they use le encoding

Comment (afilini) [reply src/wallet/mod.rs]: Looks like it's working, it looks much better this way!

**Comment by RCasatta** [comment]
utACK 6b5c3bc

**Comment by notmandatory** [comment]
utACK `6b5c3bc` , great docs, tests, and rel. notes :-)

**Pull request merged**